File: Administrativo/spring-erp-web/SpringNg/src/app/erp_module/shared/miscelaneos/servicio/MaMiscelaneosdetalleServicio.ts
import { Component, Inject, Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Header } from 'primeng/primeng';
import { MaMiscelaneosdetalle, MaMiscelaneosdetallePk } from '../dominio/MaMiscelaneosdetalle';
import { FiltroMiscelaneosDetalle } from '../dominio/filtromiscelaneosdetalle';
import { DtoTabla } from '../../dominio/dto/DtoTabla';

@Injectable()
export class MaMiscelaneosdetalleServicio {

    private rutaServicio: string;
    private http: HttpClient;
    private baseUrl: string;

    constructor(http: HttpClient, @Inject('BASE_URL') baseUrl: string) {
        this.http = http;
        this.baseUrl = baseUrl;
        this.rutaServicio = this.baseUrl + 'api/spring/core/MaMiscelaneosdetalle/';
    }

    public listarActivos(aplicacionCodigo: string, codigoTabla: string): Promise<MaMiscelaneosdetalle[]> {

        return this.http.get(this.rutaServicio + 'listarActivos/' + aplicacionCodigo + '/' + codigoTabla)
            .toPromise()
            .then(response => response as MaMiscelaneosdetalle[])
            .catch(error => []);
    }

    public listarActivosBean(aplicacionCodigo: string, codigoTabla: string): Promise<MaMiscelaneosdetalle[]> {

        return this.http.get(this.rutaServicio + 'listarActivosBean/' + aplicacionCodigo + '/' + codigoTabla)
            .toPromise()
            .then(response => response as MaMiscelaneosdetalle[])
            .catch(error => []);
    }

    public listarHeaderPorAplicacion(aplicacionCodigo: string, compania: string): Promise<DtoTabla[]> {

        return this.http.get(this.rutaServicio + 'listarHeaderPorAplicacion/' + aplicacionCodigo + '/' + compania)
            .toPromise()
            .then(response => response as DtoTabla[])
            .catch(error => []);
    }

    public obtenerPorId(pk: MaMiscelaneosdetallePk): Promise<MaMiscelaneosdetalle> {

        return this.http.post(this.rutaServicio + 'obtenerPorId', pk)
            .toPromise()
            .then(response => response as MaMiscelaneosdetalle)
            .catch(error => new MaMiscelaneosdetalle());
    }


    public listar(filtro: FiltroMiscelaneosDetalle): Promise<MaMiscelaneosdetalle[]> {

        return this.http.post(this.rutaServicio + 'listar', filtro)
            .toPromise()
            .then(response => response as MaMiscelaneosdetalle[])
            .catch(error => []);
    }

    public actualizar(bean: MaMiscelaneosdetalle): Promise<MaMiscelaneosdetalle> {
        return this.http.post(this.rutaServicio + 'actualizar', bean)
            .toPromise()
            .then(response => response as MaMiscelaneosdetalle)
            .catch(error => null);

    }

    public registrar(bean: MaMiscelaneosdetalle): Promise<MaMiscelaneosdetalle> {
        return this.http.post(this.rutaServicio + 'registrar', bean)
            .toPromise()
            .then(response => response as MaMiscelaneosdetalle)
            .catch(error => null);

    }

    public eliminar(pk: MaMiscelaneosdetallePk): Promise<any> {
        return this.http.post(this.rutaServicio + 'eliminar', pk)
            .toPromise()
            .then(response => response as any)
            .catch(error => null);

    }



}

File: Administrativo/spring-erp-servicio/net/royal/spring/core/servicio/MaMiscelaneosdetalleServicio.cs
using System;
using System.Collections.Generic;
using System.Text;

using net.royal.spring.framework.web.servicio;
using net.royal.spring.core.dominio;
using net.royal.spring.core.dominio.filtro;
using net.royal.spring.framework.core.dominio;
using net.royal.spring.framework.core.dominio.dto;

namespace net.royal.spring.core.servicio
{

public interface MaMiscelaneosdetalleServicio : GenericoServicio {
        MaMiscelaneosdetalle obtenerPorId(MaMiscelaneosdetallePk maMiscelaneosdetallePk);
        MaMiscelaneosdetalle registrar(MaMiscelaneosdetalle maMiscelaneosdetalle);
        MaMiscelaneosdetalle actualizar(MaMiscelaneosdetalle maMiscelaneosdetalle);
        void eliminar(MaMiscelaneosdetallePk maMiscelaneosdetallePk);
        List<MaMiscelaneosdetalle> listarActivos(MaMiscelaneosheaderPk maMiscelaneosheaderPk);
        List<MaMiscelaneosdetalle> listarActivos(String codigoTabla);
        List<MaMiscelaneosdetalle> listarActivos(String aplicacionCodigo, String codigoTabla);
        List<MaMiscelaneosdetalle> listarActivosBean(String aplicacionCodigo, String codigoTabla);
        
        List<MaMiscelaneosdetalle> listarActivosPorCodigo1(String aplicacionCodigo, String codigoTabla,String valorCodigo1);
        List<MaMiscelaneosdetalle> listar(FiltroMiscelaneosDetalle filtro);
        ParametroPaginacionGenerico listarPaginacion(ParametroPaginacionGenerico parametroPaginacionGenerico, MaMiscelaneosdetalle filtro);
        List<MaMiscelaneosdetalle> listarEnSentencia(FiltroMiscelaneosDetalle filtro);
        String obtenerDescripcionPorId(MaMiscelaneosdetallePk maMiscelaneosdetallePk);
        String obtenerDescripcionPorId(String aplicacion,String tabla,String elemento);
        List<MaMiscelaneosdetalle> listarDetalle(MaMiscelaneosheaderPk llave);
        List<DtoTabla> listarHeaderPorAplicacion(string aplicacionCodigo, string compania);
    }
}
